metabase/metabase
# Replace redux-form in MetricDetail form

Epic https://github.com/metabase/metabase/issues/22372

How to test:
- Go to `/admin/datamodel/metrics` and create a metric
- Go to `/reference/metrics/` and edit a metric
- Make sure it is possible to edit all fields
- Make sure that the revision message is retained

Please note that `MetricDetail` component was converted from a class to a function to use `useFormik` hook. Please ignore whitespace changes in diff to see [actual changes](https://github.com/metabase/metabase/pull/24713/files?diff=split&w=1).

Other `reference` forms would be converted in the same way in follow-up PRs.

<img width="1727" alt="Screenshot 2022-08-10 at 21 23 18" src="https://user-images.githubusercontent.com/8542534/183988358-c6efe274-0bea-4356-aa77-055537119a96.png">
<img width="1728" alt="Screenshot 2022-08-10 at 21 23 29" src="https://user-images.githubusercontent.com/8542534/183988383-37d87489-1556-4420-97d5-2006221ca555.png">
